### ISSUE TITLE
How should I obtain the field names of SQL?

### ISSUE THREAD

**Issue by xiusin** [vlang/v#20565]
### Describe the bug

like :
```vlang
mut db := sqlite.connect(':memory:')!
result := db.exec('select max(id) as max_id from db')!



```
How do I know the max_id in the field list?

**Comment by xiusin** [comment]
How do I know the max_id in the field list?

**Comment by shove70** [comment]
V don't have that yet, I think the field names are already known, and for some aggregate functions we should use "as" to determine the field names

**Comment by xiusin** [comment]
@shove70 If you want to write a custom SQL statement, it is almost impossible to get the field list directly, unless you use some analytical SQL statements to disassemble the field list

**Comment by shove70** [comment]
hmm, It's already marked as a `feature request`

**Comment by xiusin** [comment]
@shove70 MySQL can use the following function to get the list of fields:

```vlang
result := db.query('SELECT max(id) as max_id FROM jobs')!
dump(result.fields())
```

result:
```
[db.v:163] result.fields(): [
{
        name: "max_id"
        org_name: ""
        table: ""
        org_table: ""
        db: ""
        catalog: "def"
        def: ""
        length: 11
        max_length: 0
        name_length: 6
        org_name_length: 0
        table_length: 0
        org_table_length: 0
        db_length: 0
        catalog_length: 3
        def_length: 0
        flags: 32896
        decimals: 0
        charsetnr: 63
        type: long
}
]
```

**Comment by JalonSolov** [comment]
Is that a list of fields, or a description of the `max_id` field?

The proper way to show fields in a table is `SHOW COLUMNS FROM <table name>`

**Comment by shove70** [comment]
> ```v
> result := db.query('SELECT max(id) as max_id FROM jobs')!
> dump(result.fields())
> ```

Yes, currently mysql, sqlite, mssql, pg several databases have different wrappe ways for query results, sqlite query results only contain []row data, no column information.

This feature request may be scheduled for processing in due course

**Comment by xiusin** [comment]
@shove70 `C.sqlite3_column_name` I think this should be a function to get the field name, but it's not a public function

**Comment by xiusin** [comment]
@JalonSolov Most of our usage scenarios are not simply viewing the structure of a data table, but complex queries, which cannot be applied in this case.

**Comment by shove70** [comment]
> @shove70 `C.sqlite3_column_name` I think this should be a function to get the field name, but it's not a public function

Yes, it is a field name query against the query result set.
If you changed it to the pub method, you would also have to manually query the data using c.qlite3_prepare_v2, which is still inconvenient.

The best way to do this is to optimize the query result wrapper to include row and column informations.

**Comment by JalonSolov** [comment]
Or, perhaps better yet, make it support the `SHOW` statement so normal SQL can be used (and changed under the cover, if necessary, for the supported databases).

**Comment by louis77** [comment]
This PR should not be merged and instead closed. 

**Reason:**

Extracting the column names of a query result through the query itself via arbitrary regex-parsing of the query string **is wrong** and will lead to errors. In fact, `sqlite.h` already has a function that provides the proper column names of any statement and it is already used in `db.sqlite`. 99.9% if the code is already in vlib, it just needs to be exposed.
